src/lib.rs

```diff
@@ -8,10 +8,64 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-//! Utilities for random number generation
-//! 
-//! ## Example
-//! 
+//! # Getting random values
+//!
+//! To quickly get you started, the easiest and most high-level way to just get
+//! some random value is to use [`random()`].
+//!
+//! ```rust
+//! let x: u8 = rand::random();
+//! println!("{}", x);
+//!
+//! let y = rand::random::<f64>();
+//! println!("{}", y);
+//!
+//! if rand::random() { // generates a boolean
+//!     println!("Better lucky than good!");
+//! }
+//! ```
+//!
+//! This functionality is however very basic. As soon as you need something
+//! slightly more specific, like a random value in some range, you have to know
+//! a bit more about how things work, which are discussed below.
+//!
+//! ## Examples
+//!
+//! For some inspiration, see the examples:
+//!
+//! *   [Monte Carlo estimation of π](
+//!     https://github.com/rust-lang-nursery/rand/blob/master/examples/monte-carlo.rs)
+//! *   [Monty Hall Problem](
+//!     https://github.com/rust-lang-nursery/rand/blob/master/examples/monty-hall.rs)
+//!
+//!
+//! # The two-step process to get a random value
+//!
+//! To get a random value, you usually need two pieces:
+//!
+//! - a random number generator (RNG);
+//! - some function to transform the 'bag of bits' from the RNG to a value you
+//!   can use.
+//!
+//! There are many kinds of RNGs, with different trade-offs. Rand comes with a
+//! default, [`thread_rng`]. The numbers it generates are suitable to use in all
+//! situations.
```

Comment:
Perhaps say:

> There are many kinds of RNGs, but we are still able to pick a good all-use default which is fast (amortised performance), has good quality (secure and unbiased), and to automatically make a thread-local instance on first use; we call this `thread_rng`.

Reply:
I like it! Not sure about "amortised performance", I would probably expand or remove it. (BTW, are we using British or American spelling?) I think "unbiased" could be replaced with "unpredictable".